General code clean up and Feature 1 work

scheffeltravis opened the pull request:
This PR encompasses a lot of refactoring and clean up but the main things to note here are:
- Combined the search and recipe templates to simplify things
- Wired up searching to recipe database
- Added simple category search (which should probably just be a dropdown but we can do that later)

Here are some screenshots of it in action (because if I'm honest I don't know how to do a video:
1. When first logging in (you will see the edit button is only available for the logged in user):
![image](https://user-images.githubusercontent.com/9014291/99926797-80462100-2d08-11eb-946a-912873ff23b4.png)

2. Searching by title:
![image](https://user-images.githubusercontent.com/9014291/99926814-9522b480-2d08-11eb-8b50-b47aedf498b9.png)

3. Searching by category:
![image](https://user-images.githubusercontent.com/9014291/99926829-a23fa380-2d08-11eb-84c1-d1bfd103693e.png)
![image](https://user-images.githubusercontent.com/9014291/99926851-bc798180-2d08-11eb-8107-5c9786441549.png)

4. Searching by author:
![image](https://user-images.githubusercontent.com/9014291/99926857-c8654380-2d08-11eb-961c-e16d5d64f2e1.png)
![image](https://user-images.githubusercontent.com/9014291/99926866-d2874200-2d08-11eb-9f55-0d9e9f81512d.png)

The other thing to note is we don't save state in anyway so when the search POST request is made, the page reloads with the radio buttons reset.

scheffeltravis committed:
Combined search and recipe templates, clean up, wired up search to DB

CHRYSICS commented on example/flaskr/templates/recipe/create.html:
```diff
@@ -10,6 +10,14 @@ <h1>{% block title %}New Post{% endblock %}</h1>
     <textarea name="body" id="body">{{ request.form['body'] }}</textarea>
+    <label for="category">Recipe Category</label>
+    <select id="category" name="category">
+      <option value="1">Breakfast</option>
```
This is nicely done, like how it this works well for the search

CHRYSICS reviewed (approved):
Looks good to me and runs well!

CHRYSICS commented on example/flaskr/templates/recipe/index.html:
```diff
@@ -1,28 +1,51 @@
+          <h1>{{ recipe['title'] }}</h1>
+          <div class="about">by {{ recipe['username'] }} on {{ recipe['created'].strftime('%Y-%m-%d') }}</div>
         </div>
-
```
This is the only thing I would leave actually. The select will bring us to a view template where edit and save to recipe book should be there.  In F1-01, one requirement is to be able to view recipe

scheffeltravis replied:
AH, good catch. I actually think this got cleaned up when I was dealing with merge conflicts. I'll fix it up

CHRYSICS replied:
Just committed, should have fixed it :+1:

CHRYSICS committed:
leaving select for viewing recipe

This meets F1-01 requirement of viewing recipe and its ingredients, leaving the edit and saving to recipe book to recipe view template.

CHRYSICS committed:
leaving select for viewing recipe

This meets F1-01 requirement of viewing recipe and its ingredients, leaving the edit and saving to recipe book to recipe view template.

scheffeltravis committed:
Updated PR based on peer feedback

scheffeltravis committed:
Merge branch 'scheffet/code-cleanup' of https://github.com/CHRYSICS/CS361-Group-04 into scheffet/code-cleanup

scheffeltravis committed:
Correct restrictions for guest users creating recipes
